doc/developer-guides/sw_design_guidelines.rst

```diff
@@ -0,0 +1,494 @@
+==================
+
+Fulfillment to FuSa Standards
+-----------------------------
+
+The fulfillment matrix of error detection and error handling on architecture
+level to FuSa Standards [IEC_61508-7]_ in ACRN hypervisor is shown in
+:numref:`fulfillment_matrix_arch_level` below.
+
+.. table:: Fulfillment to FuSa Standards on Architecture Level
+   :align: center
+   :widths: auto
+   :name: fulfillment_matrix_arch_level
+
+   +-----------------------+---------------+----------+--------------------------------------+
+   | **Technique/Measure** | **Ref in      | **SIL3** | **Company Solution**                 |
+   |                       | IEC 61508-7** |          |                                      |
+   +=======================+===============+==========+======================================+
+   | Failure assertion     | C.3.3         | R        | This technique will be used.         |
+   | programming           |               |          | Plan to do range check in hypercalls |
+   |                       |               |          | and HW capability checks.            |
+   +-----------------------+---------------+----------+--------------------------------------+
+
+
+Error Handling Methods
+----------------------
+
+The error handling methods used in ACRN hypervisor on architecture level are
+shown below.
+
+**Invoke default fatal error handler**
+  The hypervisor shall invoke default fatal error handler when the below cases
+  occur. Customers can define platform-specific handlers, allowing them to
+  implement additional error reporting (mostly to hardware) if required. The
+  default fatal error handler will invoke platform-specific handlers defined by
+  users at first, then it will panic the system.
+
+  This method applies to the following cases:
+
+  - Related hardware resources are unavailable.
+  - Boot information is invalid during platform initialization.
+  - Unexpected exception occurs in root mode due to hardware failures.
+  - Failures occur in the VM dedicated for error handling.
+
+**Return error code**
+  The hypervisor shall return error code to the VM when the below cases occur.
+  The error code shall indicate the error type detected (e.g. invalid parameter,
+  device not found, device busy, resource unavailable, etc).
+
+  This method applies to the following case:
+
+  - The hypercall parameter from the VM is invalid.
+
+**Inform the safety VM through specific register or memory area**
+  The hypervisor shall inform the safety VM through specific register or memory
+  area when the below cases occur. The VM will decide how to handle the related
+  error. This shall only be done after the VM (Safety OS or Service OS)
+  dedicated for error handling have started.
+
+  This method applies to the following cases:
+
+  - Machine check errors occur due to hardware failures.
+
+  - Unexpected VM entry failures occur, where the VM is not the one dedicated
+    for error handling.
+
+**Crash the system via ASSERT**
```

Comment:
You used the term ``panic`` in other spots in this document... I like that better than ``crash``